src/main/kotlin/io/cloudflight/gradle/autoconfigure/node/NodeConfigurePlugin.kt

```diff
@@ -0,0 +1,142 @@
+package io.cloudflight.gradle.autoconfigure.node
+
+import com.github.gradle.node.NodeExtension
+import com.github.gradle.node.NodeExtension.Companion.DEFAULT_NPM_VERSION
+import com.github.gradle.node.NodePlugin
+import com.github.gradle.node.npm.task.NpmInstallTask
+import com.github.gradle.node.npm.task.NpmTask
+import io.cloudflight.gradle.autoconfigure.AutoConfigureGradlePlugin
+import io.cloudflight.gradle.autoconfigure.extentions.gradle.api.plugins.apply
+import io.cloudflight.gradle.autoconfigure.extentions.gradle.api.plugins.create
+import io.cloudflight.gradle.autoconfigure.java.JavaConfigurePlugin
+import io.cloudflight.gradle.autoconfigure.util.BuildUtils.isIntegrationBuild
+import io.cloudflight.gradle.autoconfigure.util.EnvironmentUtils.isVerifyBuild
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.tasks.SourceSet
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+import java.io.File
+
+class NodeConfigurePlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.plugins.apply(NodePlugin::class)
+        project.plugins.apply(JavaConfigurePlugin::class)
+
+        val node = NodeExtension[project]
+
+        // on CI environments we want to use npm ci instead of npm install
+        if (isIntegrationBuild()) {
+            node.npmInstallCommand.set("ci")
+        }
+
+        val nodeExtension = project.extensions.create(EXTENSION_NAME, NodeConfigurePluginExtension::class).apply {
+            nodeVersion.convention(NODE_VERSION)
+            downloadNode.convention(true)
+            npm.npmVersion.convention(DEFAULT_NPM_VERSION)
+            npm.destinationDir.convention(File(project.buildDir, "/generated-resources/"))
+            npm.inputFiles.convention(
+                listOf(
+                    project.fileTree("src"),
+                    project.file(NpmHelper.ANGULAR_JSON),
+                    project.file(NpmHelper.PACKAGE_JSON),
+                    project.file("tsconfig.json"),
+                    project.file("tslint.json"),
+                    project.file("eslintrc.json")
+                )
+            )
+        }
+
+
+        val dummy = NodeExtension(project.rootProject)
+        // Ensure node is only downloaded once for all subprojects
+        node.workDir.set(dummy.workDir)
+        node.npmWorkDir.set(dummy.npmWorkDir)
+        node.yarnWorkDir.set(dummy.yarnWorkDir)
+
+        node.download.set(nodeExtension.downloadNode)
+
+        project.afterEvaluate { p ->
```

Comment:
I think nothing in here needs to be done in the `afterEvaluate` because `Task.inputs.files(...)` also takes a `Provider` as argument (see [here](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#files-java.lang.Object...-)) which should be lazily evaluated. And since this seems to be the only reason why all of this is in an `afterEvaluate` we could get rid of it and allow a user to modify our tasks.